main.py

```diff
@@ -1,10 +1,65 @@
+from flask import request
+from flask import Flask, jsonify
+
 from database_connector import DatabaseConnector
 from database_manager import ItemManager, TagManager
+from search_manager import SearchManager
+from product_manager import ProductManager
 
 database_connector = DatabaseConnector('db')
 connection = database_connector.get_connection()
 item_manager = ItemManager(database_connector)
 tag_manager = TagManager(database_connector)
+search_manager = SearchManager(tag_manager, item_manager)
+product_manager = ProductManager(item_manager, tag_manager)
+
+app = Flask(__name__)
+
+
+@app.route('/')
+def index():
+    return 'Hello, World, this is the index!'
+
+
+@app.route('/unique_tags', methods=['GET', 'POST'])
+def unique_tags():
+    if request.method == 'GET':
+        return jsonify(tag_manager.get_unique_tags_list())
+    else:
+        return "no tags requested"
+
+
+@app.route('/add_product', methods=['GET', 'POST'])
+def add_product():
+    if request.method == 'GET':
+        # TODO: updated tags below to get data from FE
+        tags = ["bird", "hello"]
+        product_manager.add_product(tags)
+        return 'A product has been added'
+    else:
+        return "no button clicked yet to add a product"
+
+
+@app.route('/delete_product', methods=['GET', 'POST'])
```

Comment:
If you want to write a RESTful API, read this: https://hackernoon.com/restful-api-designing-guidelines-the-best-practices-60e1d954e7c9